### PR TITLE
Switch Azure.Core to package reference one last time

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: eng/Directory.Build.Data.props [hunk 1]
```diff
@@ -112,7 +112,6 @@
     <DelaySign>false</DelaySign>
     <PublicSign>false</PublicSign>
     <ImportDefaultReferences>false</ImportDefaultReferences>
-    <UseProjectReferenceToAzureCore>true</UseProjectReferenceToAzureCore>
```
Comment (schaabs) [inline eng/Directory.Build.Data.props]: Do we want to totally remove this switch as opposed to just setting it to false? Does it not still hold some value possibly for testing?

Comment (pakrym) [reply eng/Directory.Build.Data.props]: cc @weshaggard

Comment (pakrym) [reply eng/Directory.Build.Data.props]: I would rather we add it or something similar back when the testing story gets figured out.

Comment (pakrym) [reply eng/Directory.Build.Data.props]: Because it will have support other libraries in addition to Azure.Core like Azure.Identity and Azure.S.Blob

**Review by schaabs — APPROVED** [review]

**Comment by pakrym** [comment]
Linux hang is https://github.com/Azure/azure-sdk-for-net/issues/7634 merging to unblock releases

**Comment by weshaggard** [comment]
>Do we want to totally remove this switch as opposed to just setting it to false? Does it not still hold some value possibly for testing?

I do agree we will likely need something for testing but I also agree we don't yet know for sure what that looks like so I'm fine with removing this until we know what that is.

**Pull request merged**